datadog.go

```diff
@@ -0,0 +1,156 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// Package datadog contains a Datadog exporter.
+//
+// This exporter is currently work in progress
+// and will eventually be imported as "go.opencensus.io/exporter/datadog"
+package datadog
+
+import (
+	"bytes"
+	"errors"
+	"log"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+const (
+	defaultHost   = "localhost"
+	defaultPort   = "8125"
+	opencensusTag = "source:Opencensus"
+)
+
+var (
+	newExporterOnce      sync.Once
+	errSingletonExporter = errors.New("expecting only one exporter per instance")
+	err                  = errSingletonExporter
+	exporter             *Exporter
+	tags                 = []string{opencensusTag}
+	reg                  = regexp.MustCompile("[^a-zA-Z0-9]+")
+)
+
+// Exporter exports stats to Datadog
+type Exporter struct {
+	opts      Options
+	collector *collector
+}
+
+// ExportView exports to Datadog if view data has one or more rows.
+func (e *Exporter) ExportView(vd *view.Data) {
+	if len(vd.Rows) == 0 {
+		return
+	}
+	e.collector.addViewData(vd)
+}
+
+// Options contains options for configuring the exporter
+type Options struct {
+	// Namespace to prepend to all metrics
+	Namespace string
+
+	// Host for DogStatsD connection
```

Comment:
Originally I had that as a string, but it was suggested to separate them into `host` and `port` for better error handling. It would also follow the DogStatsD configuration format in the agent.